test/src/System/Vector2.cpp

```diff
@@ -145,4 +146,47 @@ TEST_CASE("sf::Vector2 class template", "[system]")
             CHECK_FALSE(firstEqualVector != secondEqualVector);
         }
     }
+
+    SECTION("Structured bindings")
+    {
+        sf::Vector2i vector(1, 2);
+
+        SECTION("destructure by value")
+        {
+            auto [x, y] = vector;
+
+            CHECK(x == 1);
+            CHECK(y == 2);
+
+            static_assert(std::is_same_v<decltype(x), decltype(vector.x)>);
+
+            x = 3;
+
+            CHECK(x == 3);
+            CHECK(vector.x == 1);
+        }
+
+        SECTION("destructure by ref")
+        {
+            auto& [x, y] = vector;
+
+            CHECK(x == 1);
+            CHECK(y == 2);
+
+            static_assert(std::is_same_v<decltype(x), decltype(vector.x)>);
+
+            x = 3;
+
+            CHECK(x == 3);
+            CHECK(vector.x == 3);
+        }
+    }
+
+    SECTION("Constexpr support")
+    {
+        constexpr sf::Vector2i vector(1, 2);
+        static_assert(vector.x == 1);
+        static_assert(vector.y == 2);
+        static_assert(vector + sf::Vector2i(2, 1) == sf::Vector2i(3, 3));
```

Comment:
Once we've updated Catch2 in #1898, we could also use `STATIC_REQUIRE`

Reply:
Is there any practical benefit over `static_assert`?

Reply:
It's properly tracked as Catch2 assertion. Currently the test will just fail because of C++, but Catch2 won't really "know" about it.

See https://github.com/catchorg/Catch2/issues/1356

Reply:
I see, I am happy to update in a future PR.